### PR TITLE
[Enhancement] Add session variable query_mem_limit

### PR DESCRIPTION
**Pull request by ZiheLiu**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Problem Summary：

`exec_mem_limit` is a little bit misunderstanding, since the max memory of a query is not `exec_mem_limit` but `exec_mem_limit * num_fragments * pipeline_dop`, and fragments/pipeline is our internal implementation concept.

Therefore, add a new session variable `query_mem_limit`.
- If `query_mem_limit` > 0, use it to limit the memory of a query.
   The memory a query able to be used is just `query_mem_limit`.
- Otherwise, use `exec_mem_limit` to limit the memory of a query.

To maintain compatibility, the default value is 0.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 15 / 16 (93.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/analysis/SetVar.java|12|12|100.00%|

**Comment by imay** [comment]
@Dshadowzh 
You should check if we need this variable and if this variable name is OK.

**Comment by ZiheLiu** [comment]
run starrocks_fe_unittest

**Comment by ZiheLiu** [comment]
run starrocks_fe_unittest

**Comment by ZiheLiu** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 15 / 16 (93.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/analysis/SetVar.java|12|12|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 1]
```diff
@@ -47,6 +47,28 @@ public class SessionVariable implements Serializable, Writable, Cloneable {
...
+     * This variable is mainly for INSERT operation, because INSERT operation has both query and load part.
+     * Using only the exec_mem_limit variable does not make a good distinction of memory limit between the two parts.
+     */
+    public static final String LOAD_MEM_LIMIT = "load_mem_limit";
+
+    /**
+     * The mem limit of query on BE. It takes effects only when enabling pipeline engine.
+     * - If `query_mem_limit` > 0, use it to limit the memory of a query.
+     *   The memory a query able to be used is just `query_mem_limit`.
+     * - Otherwise, use `exec_mem_limit` to limit the memory of a query.
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:66]: Maybe in future version, we should remove the limit if `query_mem_limit == 0` , but not use `exec_mem_list` as limit?

**Review by murphyatwork — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java (hunk 1 above)
Comment (ZiheLiu) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:66]: Maybe for the compatibility, we can not remove `exec_mem_list` easily.

**Review by kangkaisen — DISMISSED** [review]

**Comment by kangkaisen** [comment]
You need to add doc for this PR. and let our users know how to set default value for query_mem_limit

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 15 / 16 (93.75%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/analysis/SetVar.java|12|12|100.00%|

**Review by kangkaisen — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Pull request merged**